zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/NotebookAuthorization.java

```diff
@@ -133,27 +115,14 @@ public void setRoles(String user, Set<String> roles) {
   }
   
   private void saveToFile() {
-    String jsonString;
-
     synchronized (authInfo) {
       NotebookAuthorizationInfoSaving info = new NotebookAuthorizationInfoSaving();
       info.authInfo = authInfo;
-      jsonString = gson.toJson(info);
-    }
-
-    try {
-      File settingFile = new File(filePath);
-      if (!settingFile.exists()) {
-        settingFile.createNewFile();
+      try {
+        configStorage.save(info);
+      } catch (IOException e) {
+        LOG.error("Error saving notebook authorization file", e);
```

Comment:
recommend adding `filePath` to that error string

Reply:
The storage file path will be logged in `FileSystemConfigStorage`. And it doesn't expose api for getting the config file path. Actual `ConfigStorage` is a generic interface, it doesn't need to store into a file, the storage could be other storage like DB and etc.

Reply:
OK. I'm just worrying about fielding the support calls where all you get is that stack trace.